sycl/test/basic_tests/is_device_copyable.cpp

```diff
@@ -0,0 +1,78 @@
+// RUN: %clangxx -fsycl -fsycl-device-only -fsyntax-only -Xclang -verify \
+// RUN: %s -I %sycl_include
+
+// This test checks that compiler does not report any errors
+// for the device copyable types being passed from HOST to DEVICE.
+// It also checks that the types not copyable in general, but declared as
+// device copyable (by using specialization of is_device_copyable class)
+// are allowed.
+
+// expected-no-diagnostics
+
+#include <CL/sycl.hpp>
+
+using namespace cl::sycl;
+
+struct A {
+  int i;
+};
+
+// Not copyable type, but it will be declared as device copyable.
+struct BCopyable {
+  int i;
+  BCopyable(int _i) : i(_i) {}
+  BCopyable(const BCopyable &x) : i(x.i) {}
+};
+
+struct C : A {
+  const A C2;
```

Comment:
It'd be useful to have some comments around here explaining that this type is not trivially copyable but it is trivially copy constructible and so this hits the deprecated functionality (right?).

Reply:
Added in https://github.com/intel/llvm/pull/3905/commits/da49819bce2df28310ba8af73934aa87fcf88ce6